docroot/modules/custom/foia_webform/src/Plugin/QueueWorker/FoiaSubmissionQueueWorker.php

```diff
@@ -73,8 +75,46 @@ public function processItem($data) {
     $agencyComponent = Node::load($agencyComponentId);
     $submissionService = $this->foiaSubmissionServiceFactory->get($agencyComponent);
 
+    $foiaRequest->set('field_submission_time', REQUEST_TIME);
     // Submit the form values to the Agency Component.
     $validSubmissionResponse = $submissionService->sendRequestToComponent($foiaRequest, $agencyComponent);
+
+    if ($validSubmissionResponse) {
+      $foiaRequest->setRequestStatus(FoiaRequestInterface::STATUS_SUBMITTED);
+      $submissionMethod = isset($validSubmissionResponse['type']) ? $validSubmissionResponse['type'] : '';
+      $responseCode = isset($validSubmissionResponse['response_code']) ? $validSubmissionResponse['response_code'] : '';
+      $caseManagementId = isset($validSubmissionResponse['id']) ? $validSubmissionResponse['id'] : '';
+      $caseManagementStatusTrackingNumber = isset($validSubmissionResponse['status_tracking_number']) ? $validSubmissionResponse['status_tracking_number'] : '';
+      if ($caseManagementId) {
+        $foiaRequest->set('field_case_management_id', $caseManagementId);
+      }
+      if ($caseManagementStatusTrackingNumber) {
+        $foiaRequest->set('field_tracking_number', $caseManagementStatusTrackingNumber);
+      }
+      $webformSubmissionId = $foiaRequest->get('field_webform_submission_id')->value;
+      $webformSubmission = WebformSubmission::load($webformSubmissionId);
+      $webformSubmission->delete();
+      // @todo change http_code in service api error to response_code
+    }
+    else {
+      $foiaRequest->setRequestStatus(FoiaRequestInterface::STATUS_FAILED);
+      $invalidSubmissionInfo = $submissionService->getSubmissionErrors();
+      $submissionMethod = isset($invalidSubmissionInfo['type']) ? $invalidSubmissionInfo['type'] : '';
+      // @todo change http_code in service api error to response_code
```

Comment:
Looks like this `todo` comment and the one on line 97 have already been taken care of

Reply:
Fixed in a subsequent commit